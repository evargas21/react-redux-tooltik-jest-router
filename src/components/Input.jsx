// @packages
import * as ProtoTypes from 'prop-types';

const Input =({ disabled, label, name, onChange, type, value }) => {
    return (
        <div key={name}>
            <label htmlFor={name}>{label}</label>
            <input
                className="custom-input"
                value={value}
                type={type}
                onChange={onChange}
                name={name}
                id={name}
                placeholder={label}
                disabled={disabled}
                required
                aria-autocomplete="none"
            />
        </div>
    );
};

Input.defaulttProps = {
    disabled: false,
    label: '',
    type: 'text',
    value: ''
};

Input.propTypes = {
    disabled: ProtoTypes.bool,
    label: ProtoTypes.string,
    name: ProtoTypes.string.isRequired,
    onChange: ProtoTypes.func.isRequired,
    type: ProtoTypes.string,
    value: ProtoTypes.string
};

export default Input;
