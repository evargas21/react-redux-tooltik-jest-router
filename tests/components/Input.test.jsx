// @packages
import { fireEvent, render, screen } from '@testing-library/react';

// @scripts
import Input from '../../src/components/Input';

describe('Test in <Input />', () => {
    const name = "test";

    test('must match snapshot', () => {
        const { container } = render(<Input name={name} onChange={jest.fn()} value={name}/>);
        expect(container).toMatchSnapshot();
    });

    test('must display the input with the indicated value and disabled', () => {
        render(<Input name={name} onChange={jest.fn()} value={name} disabled />);
        const { value, disabled } = screen.getByRole('textbox');
        expect(value).toBe(name);
        expect(disabled).toBeTruthy();
    });
});
