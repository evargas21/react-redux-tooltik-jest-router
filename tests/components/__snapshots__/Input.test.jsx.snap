// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Input /> must match snapshot 1`] = `
<div>
  <div>
    <label
      for="test"
    />
    <input
      aria-autocomplete="none"
      class="custom-input"
      id="test"
      name="test"
      required=""
      value="test"
    />
  </div>
</div>
`;
